### PR TITLE
Edits for BOLT-07

### PR DESCRIPTION
**Pull request by shannona**
A variety of edits for clarity and our styles sheets for BOLT-7. The one other issue for BOLT-7 has been filed as issue #302.

### CODE REVIEW

File: 07-routing-gossip.md [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
+channels between two nodes. They can also exchange `channel_update`
+messages, which update information about a channel. There can only be
+one valid `channel_announcement` for any channel, but at least two
+`channel_update` messages are expected.
 
-There can only be one valid `channel_announcement` for any channel,
-but multiple `node_announcement` messages are possible (to update node
-information), and at least two `channel_update` messages are expected.
+To support node discovery, peers exchange `node_announcement`
+messages, which supply additional information about nodes There can be
```
Comment (rustyrussell) [inline 07-routing-gossip.md]: Missing period between "about nodes" and "There can be"

File: 07-routing-gossip.md [hunk 2]
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
...
 
 The `short_channel_id` is the unique description of the funding transaction.
-It is constructed with the most significant 3 bytes as the block
+It is constructed with the most significant 3 bytes indicating the block
 height, the next 3 bytes indicating the transaction index within the
 block, and the least significant two bytes indicating the output
-index which pays to the channel.
+index that pays to the channel.
+
+If the `open_channel` message had the `announce_channel` bit set, then both nodes must send the `announcement_signatures` message, otherwise they MUST NOT.
```
Comment (rustyrussell) [inline 07-routing-gossip.md]: must -> MUST.

File: 07-routing-gossip.md [hunk 3]
```diff
@@ -474,13 +479,13 @@ channel:
...
+Also, each node has a set fee scheme that it uses for each of its
 channels:
 
 1. A: 100 base + 1000 millionths
-1. B: 200 base + 2000 millionths
-1. C: 300 base + 3000 millionths
-1. D: 400 base + 4000 millionths
+2. B: 200 base + 2000 millionths
+3. C: 300 base + 3000 millionths
+4. D: 400 base + 4000 millionths
```
Comment (rustyrussell) [inline 07-routing-gossip.md]: This is fine, but FYI, markdown turns these into an enumerated list when rendering anyway :)

File: 07-routing-gossip.md [hunk 4]
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
 
 ### Requirements
 
-If the `open_channel` message had the `announce_channel` bit set, then both nodes must send the `announcement_signatures` message, otherwise they MUST NOT.
-
-If sent, `announcement_signatures` messages MUST NOT be sent until `funding_locked` has been sent and the funding transaction has at least 6 confirmations.
+The `announcement_signatures` message is created by constructing a `channel_announcement` message corresponding to the newly established channel and signing it with the secrets matching an endpoint's `node_id` and `bitcoin_key`. The message is then sent using an `announcement_signatures`.
```
Comment (toadlyBroodle) [inline 07-routing-gossip.md]: ...message, corresponding...channel, and...

File: 07-routing-gossip.md [hunk 5]
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
...
-It is constructed with the most significant 3 bytes as the block
+It is constructed with the most significant 3 bytes indicating the block
 height, the next 3 bytes indicating the transaction index within the
 block, and the least significant two bytes indicating the output
-index which pays to the channel.
+index that pays to the channel.
+
+If the `open_channel` message had the `announce_channel` bit set, then both nodes must send the `announcement_signatures` message, otherwise they MUST NOT.
+
+If sent, `announcement_signatures` messages MUST NOT be sent until `funding_locked` has been sent and the funding transaction has at least 6 confirmations.
```
Comment (toadlyBroodle) [inline 07-routing-gossip.md]: six

File: 07-routing-gossip.md [hunk 6]
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
...
 The channel is not really usable until at least one side has announced
 its fee levels and expiry using `channel_update`.
 
-In order to prove the existence of channel between `node_1` and
-`node_2` we need to:
+Proving the existence of a channel between `node_1` and
+`node_2` requires:
 
-1. Prove that the funding transaction pays to `bitcoin_key_1` and
+1. Proving that the funding transaction pays to `bitcoin_key_1` and
```
Comment (toadlyBroodle) [inline 07-routing-gossip.md]: proving...`bitcoin_key_1`  <--(no 'and')

File: 07-routing-gossip.md [hunk 7]
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
...
    `bitcoin_key_2`.
-2. Prove `node_1` owns `bitcoin_key_1`
-3. Prove `node_2` owns `bitcoin_key_2`
+2. Proving that `node_1` owns `bitcoin_key_1`
+3. Proving that `node_2` owns `bitcoin_key_2`
```
Comment (toadlyBroodle) [inline 07-routing-gossip.md]: proving...

File: 07-routing-gossip.md [hunk 8]
```diff
@@ -28,49 +34,50 @@ The willingness of the initiating node to announce the channel is signaled durin
...
-The second two are done by explicit signatures (`bitcoin_signature_1`
+The second two proofs are accomplished through explicit signatures (`bitcoin_signature_1`
 and `bitcoin_signature_2`, generated by each `bitcoin_key` and signing
-the corresponding `node_id`.
+the corresponding `node_id`).
 
-We also need to prove that `node_1` and `node_2` both agree on this
-announcement message; that is done by having a signature from each
+It is also necessary to prove that `node_1` and `node_2` both agree on this
+announcement message; that is accomplished by having a signature from each
```
Comment (toadlyBroodle) [inline 07-routing-gossip.md]: ...message; this is...

File: 07-routing-gossip.md [hunk 9]
```diff
@@ -256,50 +263,51 @@ if `port` is zero.
...
 Future fields beyond `addresses` can still be added, optionally with
 padding within `addresses` if they require certain alignment.
 
 ## The `channel_update` message
 
 After a channel has been initially announced, each side independently
 announces the fees and minimum expiry delta it requires to relay HTLCs
-through this channel.  It uses the 8-byte
-channel shortid which matches the `channel_announcement` and one bit
+through this channel. They use the 8-byte
```
Comment (toadlyBroodle) [inline 07-routing-gossip.md]: ...channel. Each uses the...

File: 07-routing-gossip.md [hunk 10]
```diff
@@ -256,50 +263,51 @@ if `port` is zero.
...
+channel shortid that matches the `channel_announcement` and one bit
```
Comment (toadlyBroodle) [inline 07-routing-gossip.md]: ... and 1 bit

File: 07-routing-gossip.md (hunk 1 above)
Comment (shannona) [reply 07-routing-gossip.md]: Fixed.

File: 07-routing-gossip.md (hunk 2 above)
Comment (shannona) [reply 07-routing-gossip.md]: Fixed.

File: 07-routing-gossip.md (hunk 4 above)
Comment (shannona) [reply 07-routing-gossip.md]: Yep, clearer. Fixed.

File: 07-routing-gossip.md (hunk 5 above)
Comment (shannona) [reply 07-routing-gossip.md]: Fixed.

File: 07-routing-gossip.md (hunk 6 above)
Comment (shannona) [reply 07-routing-gossip.md]: The whole line is actually: "1. Proving that the funding transaction pays to `bitcoin_key_1` and
   `bitcoin_key_2`."Unless there's a style I'm missing, kept.

File: 07-routing-gossip.md (hunk 7 above)
Comment (shannona) [reply 07-routing-gossip.md]: I think I didn't have that style update when I did 07, so keep an eye out when you get there. In any case, I've fixed all three of these lines to be no initial cap and no end period,

File: 07-routing-gossip.md (hunk 8 above)
Comment (shannona) [reply 07-routing-gossip.md]: Done.

File: 07-routing-gossip.md (hunk 9 above)
Comment (shannona) [reply 07-routing-gossip.md]: Changed.

File: 07-routing-gossip.md (hunk 10 above)
Comment (shannona) [reply 07-routing-gossip.md]: Fixed.

File: 07-routing-gossip.md (hunk 7 above)
Comment (shannona) [reply 07-routing-gossip.md]: (Actually, I reskimmed, and this is the only list with fragments, so we should have this now.

**Comment by shannona** [comment]
Updated commit: (1) addresses all reviews from @rustyrussell and @toadlyBroodle ; (2) Correctly capitalizes a few sections that had been missed; and (3) incorporates table of contents courtesy of @bcongdon in #310

**Pull request merged**